public/app/plugins/panel/graph3/module.tsx

```diff
@@ -0,0 +1,180 @@
+              { value: 5, label: '5' },
+              { value: 6, label: '6' },
+              { value: 7, label: '7' },
+              { value: 8, label: '8' },
+              { value: 9, label: '9' },
+              { value: 10, label: '10 • thick' },
+            ],
+          },
+          showIf: c => c.points.show,
+        })
+        .addBooleanSwitch({
+          path: 'bars.show',
+          name: 'Show bars',
+          description: '',
+          defaultValue: false,
+        })
+        .addSelect({
+          path: 'fill.alpha',
+          name: 'Fill area opacity',
+          defaultValue: 0,
+          settings: {
+            options: [
+              { value: 0, label: 'No Fill' },
+              { value: 0.1, label: '10% • transparent' },
+              { value: 0.2, label: '20%' },
+              { value: 0.3, label: '30%' },
+              { value: 0.4, label: '40% ' },
+              { value: 0.5, label: '50%' },
+              { value: 0.6, label: '60%' },
+              { value: 0.7, label: '70%' },
+              { value: 0.8, label: '80%' },
+              { value: 0.9, label: '90%' },
+              { value: 1, label: '100% • opaque' },
+            ],
+          },
+        })
+        .addTextInput({
+          path: 'axis.label',
+          name: 'Axis Label',
+          category: ['Axis'],
+          defaultValue: '',
+          settings: {
+            placeholder: 'Optional text',
+          },
+          // no matter what the field type is
+          shouldApply: () => true,
+        })
+        .addRadio({
+          path: 'axis.side',
+          name: 'Y axis side',
+          category: ['Axis'],
+          defaultValue: 3,
+          settings: {
+            options: [
+              { value: 3, label: 'Left' },
+              { value: 1, label: 'Right' },
+            ],
+          },
+        })
+        .addNumberInput({
+          path: 'axis.width',
+          name: 'Y axis width',
+          category: ['Axis'],
+          defaultValue: 60,
+          settings: {
+            placeholder: '60',
+          },
+        })
+        .addBooleanSwitch({
+          path: 'axis.grid',
+          name: 'Show axis grid',
+          category: ['Axis'],
+          description: '',
+          defaultValue: true,
+        });
+    },
+  })
+  .setPanelOptions(builder => {
+    builder
+      .addRadio({
+        path: 'tooltipOptions.mode',
+        name: 'Tooltip mode',
+        description: '',
+        defaultValue: 'single',
+        settings: {
+          options: [
+            { value: 'single', label: 'Single series' },
+            { value: 'multi', label: 'All series' },
+            { value: 'none', label: 'No tooltip' },
+          ],
+        },
+      })
+      .addBooleanSwitch({
+        path: 'graph.realTimeUpdates',
+        name: 'Real time updates',
+        description: 'continue to update the graph so the time axis matches the clock.',
+        defaultValue: false,
+      })
```

Comment:
do we need this? feels like we only need this for streaming data and if could detect that somehow

Reply:
and streaming data should have an update request time range